pkg/provider/wif_configs.go

```diff
@@ -34,13 +37,38 @@ func getWifConfigs(client *cmv1.Client) (wifConfigs []*cmv1.WifConfig, err error
 }
 
 func GetWifConfigs(client *cmv1.Client) (wifConfigs []*cmv1.WifConfig, err error) {
-	return getWifConfigs(client)
+	return getWifConfigs(client, "")
+}
+
+// GetUnusedWifConfigs returns the WIF configurations that are not associated with any cluster
+func GetUnusedWifConfigs(client *cmv1.Client) (wifConfigs []*cmv1.WifConfig, err error) {
+	return getWifConfigs(client, unusedWifsQuery)
+}
+
+// GetWifConfig returns the WIF configuration where the key is the wif config id or name
+func GetWifConfig(client *cmv1.Client, key string) (wifConfig *cmv1.WifConfig, err error) {
+	query := fmt.Sprintf(
+		"id = '%s' or display_name = '%s'",
+		key, key,
+	)
+	wifs, err := getWifConfigs(client, query)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(wifs) == 0 {
+		return nil, fmt.Errorf("WIF configuration with identifier or name '%s' not found", key)
+	}
+	if len(wifs) > 1 {
+		return nil, fmt.Errorf("there are %d WIF configurations found with identifier or name '%s'", len(wifs), key)
+	}
+	return wifs[0], nil
 }
 
 // GetWifConfigNameOptions returns the wif config options for the cluster
 // with display name as the value and id as the description
 func GetWifConfigNameOptions(client *cmv1.Client) (options []arguments.Option, err error) {
-	wifConfigs, err := getWifConfigs(client)
+	wifConfigs, err := getWifConfigs(client, unusedWifsQuery)
```

Comment:
Minor, but I was wondering if it would be better for readability to call GetUnusedWifConfigs here?

Reply:
Changed to GetUnusedWifConfigs